libs/can/include/CANMsg.h

```diff
@@ -37,62 +37,41 @@ class CANMsg : public CANMessage {
   /** Sets the ID for a CAN messages
    */
   void setID(const HWBRIDGE::CANID newID) {
-    id = static_cast<uint16_t>(newID);
+    this->id = static_cast<uint16_t>(newID);
   }
 
   /** Returns the ID of the CAN message
    */
   HWBRIDGE::CANID getID() const {
-    return static_cast<HWBRIDGE::CANID>(id);
+    return static_cast<HWBRIDGE::CANID>(this->id);
   }
 
   /** Clears CAN message content
    */
   void clear(void) {
-    len    = 0;
-    type   = CANData;
-    format = CANStandard;
-    id     = 0;
-    memset(data, 0, 8);
+    this->len    = 0;
+    this->type   = CANData;
+    this->format = CANStandard;
+    this->id     = 0;
+    memset(this->data, 0, 8);
   };
 
   /** Set the payload data
    */
   template <class T>
   void setPayload(const T value) {
-    MBED_ASSERT_WARN(sizeof(T) <= 8);  // Make sure input type fits
+    MBED_ASSERT(sizeof(T) <= 8);  // Make sure input type fits
     CANPayload<T> *payload = (CANPayload<T> *)&data;
     payload->value         = value;
-    len                    = sizeof(T);
+    this->len              = sizeof(T);
   }
 
   /** Get the payload data
    */
   template <class T>
   void getPayload(T &val) {
-    MBED_ASSERT_WARN(len = sizeof(T));  // Make sure output fits
+    MBED_ASSERT(this->len = sizeof(T));  // Make sure output fits
     CANPayload<T> *payload = (CANPayload<T> *)&data;
     val                    = payload->value;
   }
-
-  /** Append operator: Appends data (value) to CAN message
-   */
-  template <class T>
-  CANMsg &operator<<(const T val) {
-    MBED_ASSERT_WARN(len + sizeof(T) <= 8);
-    *reinterpret_cast<T *>(&data[len]) = val;
-    len += sizeof(T);
-    return *this;
-  }
-
-  /** Extract operator: Extracts data (value) from CAN message
-   */
-  template <class T>
-  CANMsg &operator>>(T &val) {
-    MBED_ASSERT_WARN(sizeof(T) <= len);
-    val = *reinterpret_cast<T *>(&data[0]);
-    len -= sizeof(T);
-    memcpy(data, data + sizeof(T), len);
-    return *this;
-  }
```

Comment:
deleted these stream overloads since we never used them anyways. I struggle to find a use for these especially since can msgs need to be structured in a very specific way. streaming just doesnt make sense.